runtime-attach/README.md

```diff
@@ -1,3 +1,29 @@
+# Runtime attachment
+
+If you can't update the JVM arguments to attach the [OpenTelemetry Java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation) (_-javaagent:path/to/opentelemetry-javaagent.jar_), this project allows you to do attachment programmatically.
+
+The `io.opentelemetry.contrib.attach.RuntimeAttach` class has an `attachJavaagentToCurrentJVM` method allowing to trigger the attachment of the OTel agent for Java.
+
+The attachment will not be initiated in the following cases:
+* The `otel.javaagent.enabled` property is set to `false`
+* The `OTEL_JAVAAGENT_ENABLED` environment variable is set to `false`
+* The attachment is not requested from the _main_ thread
```

Comment:
what do you think about adding one more check, that the call stack only has `main` in it, i.e. that it was really called directly from the initial `main` method?

Reply:
Implemented. The new code be tested with the following class:
``` java
public class MainClass {

  public static void main(String[] args) {
    System.out.println("isMainMethod = " + isMainMethod());
  }

  static boolean isMainMethod() {
    StackTraceElement bottomOfStack = findBottomOfStack(Thread.currentThread());
    String methodName = bottomOfStack.getMethodName();
    return "main".equals(methodName);
  }

  private static StackTraceElement findBottomOfStack(Thread thread) {
    StackTraceElement[] stackTrace = thread.getStackTrace();
    return  stackTrace[stackTrace.length - 1];
  }
}
```

Reply:
I think typically the `main` thread will always be rooted at the `main()` method(?), I was thinking of a more restrictive chck that `attach()` was called directly from `main()`

Reply:
I'm going to merge this for the 1.15.0 release, and we can discuss / follow-up on this point later